File: HerbalLife/src/main/java/com/example/HerbalLife/service/OrderService.java
package com.example.HerbalLife.service;

import java.util.List;

import com.example.HerbalLife.entity.Order;

public interface OrderService {

    Order placeOrder(Order order);

    List<Order> getOrders(String email);

    List<Order> getAllOrders();

    Order updateStatus(String id, String status);

}

File: HUI2/src/services/OrderService.js
import axios from "axios";

const API_URL = "http://localhost:8082/orders";

export const placeOrder = (order) => {
    return axios.post(API_URL, order);
};

export const getOrders = (email) => {
    return axios.get(`${API_URL}/${email}`);
};

export const getAllOrders = () => {
    return axios.get(`${API_URL}/all`);
};

export const updateOrderStatus = (id, status) => {
    return axios.put(`${API_URL}/${id}/status`, { status });
};